src/sentry/profiles/flamegraph.py

```diff
@@ -0,0 +1,58 @@
+from datetime import datetime, timedelta
+from typing import Dict, List
+
+from snuba_sdk import Column, Condition, Entity, Op, Query, Request
+
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.snuba.referrer import Referrer
+from sentry.utils.snuba import raw_snql_query
+
+MAX_RETENTION_DAYS = 90
+
+
+def get_profiles_id(
```

Comment:
We should look at using the existing `sentry.search.events.builder.QueryBuilder` to construct the query. This will allow us to pass the user search conditions through to the flamegraph endpoint and update the flamegraph accordingly. e.g. the user can add something like `organization.slug:sentry` and the query builder is able parse then construct the necessary condition for the query for us